### PR TITLE
DOC: update the str_cat() docstring (Delhi) 

### PR DESCRIPTION
**Pull request by vipinkjonwal**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
# paste output of "scripts/validate_docstrings.py <your-function-or-method>" here
"Docstring for "pandas.Series.str.cat" correct. :)"
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Please show the output of the validation script as asked

File: pandas/core/strings.py [hunk 1]
```diff
@@ -51,11 +51,19 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
     """
     Concatenate strings in the Series/Index with given separator.
 
+    This function concatenates elements of Series/Index and elements
+    of lists or list-like objects having same length.
+    The concatenation is done at corresponding indices of
+    iterable specified by `Series` when two or more iterables
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: This part is not fully clear to me: " corresponding indices of iterable"

Can you give a small example of what you want to explain here?

File: pandas/core/strings.py [hunk 2]
```diff
@@ -68,29 +76,29 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
     When ``na_rep`` is `None` (default behavior), NaN value(s)
     in the Series are ignored.
 
-    >>> Series(['a','b',np.nan,'c']).str.cat(sep=' ')
+    >>> pd.Series(['a','b',np.nan,'c']).str.cat(sep=' ')
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: Can you split this example in two steps?

s = pd.Series(...) 
s.str.cat(sep=..)

File: pandas/core/strings.py [hunk 3]
```diff
@@ -68,29 +76,29 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
     When ``na_rep`` is `None` (default behavior), NaN value(s)
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: I would also say something here about not passing `other` that then all values in the Series are concatenated in a single string (so to indicate you first give an example of this part of the behaviour of the method)

File: pandas/core/strings.py [hunk 4]
```diff
@@ -51,11 +51,19 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
...
+    This function concatenates elements of Series/Index and elements
+    of lists or list-like objects having same length.
+    The concatenation is done at corresponding indices of
+    iterable specified by `Series` when two or more iterables
+    are present. The final concatenation is done with a given `sep`
+    and a string is returned.
+
     Parameters
     ----------
     others : list-like, or list of list-likes
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: Can you add ``, optional`` at the end of the line?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added some first comments

File: pandas/core/strings.py (hunk 1 above)
Comment (vipinkjonwal) [reply pandas/core/strings.py]: Corresponding indices indicates that the concatenation is being done in one to one correspondence with the indices i.e, index of iterable, let say list.

For example :
>>> pd.Series(['a', 'b', 'c']).str.cat(['A', 'B', 'C'], sep=',')
    index
    0             a,A (a with A i.e, 0th index of list in Series and 0th index of list in cat(list...) ) 
    1             b,B (b with B)
    2             c,C (c with C)

File: pandas/core/strings.py (hunk 2 above)
Comment (vipinkjonwal) [reply pandas/core/strings.py]: Yes, I can do this way as well.

File: pandas/core/strings.py (hunk 3 above)
Comment (vipinkjonwal) [reply pandas/core/strings.py]: Added :-)

File: pandas/core/strings.py (hunk 4 above)
Comment (vipinkjonwal) [reply pandas/core/strings.py]: Yes. Adding.

File: pandas/core/strings.py [hunk 5]
```diff
@@ -51,46 +51,61 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
...
     na_rep : string or None, default None
         If None, NA in the series are ignored.
 
     Returns
     -------
     concat : Series/Index of objects or str
 
+    See Also
+    --------
+    str_split : Split each string in the Series/Index
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: str_split -> split (`str_split` is an implementation detail in this case, `split`is the user-facing interface)

File: pandas/core/strings.py [hunk 6]
```diff
@@ -51,46 +51,61 @@ def str_cat(arr, others=None, sep=None, na_rep=None):
...
+
     Examples
     --------
     When ``na_rep`` is `None` (default behavior), NaN value(s)
     in the Series are ignored.
 
-    >>> Series(['a','b',np.nan,'c']).str.cat(sep=' ')
+    >>> s = pd.Series(['a','b',np.nan,'c'])
```
Comment (jorisvandenbossche) [inline pandas/core/strings.py]: Can you add spaces after the comma (PEP8)

(same for the examples below)

File: pandas/core/strings.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/core/strings.py]: Ah, OK. In that case, I think it is fine to say that "concatenates the Series/Index and `other` element-wise"

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the updates! Added some additional comments

File: pandas/core/strings.py (hunk 3 above)
Comment (jorisvandenbossche) [reply pandas/core/strings.py]: Did you add this?

**Review by jorisvandenbossche — APPROVED** [review]
Simplified the extended summary a bit (see my feedback), looks good for the rest!

**Comment by jorisvandenbossche** [comment]
@vipinkjonwal Thanks for the PR!

**Pull request merged**